lib/matplotlib/cbook/__init__.py

```diff
@@ -2767,3 +2767,44 @@ def _topmost_artist(
     in reverse order.
     """
     return _cached_max(reversed(artists))
+
+
+def _define_aliases(alias_d, cls=None):
+    """Class decorator for defining property aliases.
+
+    Use as ::
+
+        @cbook._define_aliases({"property": ["alias", ...], ...})
+        class C: ...
+
+    For each property, if the corresponding ``get_property`` is defined in the
+    class so far, an alias named ``get_alias`` will be defined; the same will
+    be done for setters.  If neither the getter nor the setter exists, an
+    exception will be raised.
+
+    The alias map is stored as the ``_alias_map`` attribute on the class and
+    can be used by `~.normalize_kwargs`.
+    """
+    if cls is None:
+        return functools.partial(_define_aliases, alias_d)
+
+    def make_alias(name):  # Enfore a closure over *name*.
+        def method(self, *args, **kwargs):
+            return getattr(self, name)(*args, **kwargs)
+        return method
+
+    for prop, aliases in alias_d.items():
+        exists = False
+        for prefix in ["get_", "set_"]:
+            if prefix + prop in vars(cls):
+                exists = True
+                for alias in aliases:
+                    method = make_alias(prefix + prop)
+                    method.__name__ = prefix + alias
+                    method.__doc__ = "alias for `{}`".format(prefix + prop)
+                    setattr(cls, prefix + alias, method)
+        if not exists:
+            raise ValueError("property {} does not exist".format(prop))
+
+    cls._alias_map = alias_d
```

Comment:
I might be tempted to have `_define_aliases` read this, rather than include it in the decorator argument - that means the aliases come after the definitions too.

Reply:
I really would prefer the aliases to stay close to the decorator, at least.

Reply:
I prefer it this way. to keep the decorator and the alias map close together.

Would it make sense to have the decorator also add a `_normalize_kw` method as

```python
def normalize_kvwargs(self, kw):
    return normalize_kwargs(kw, self._alias_map)
```

?

Might also be better to check if the class already has an `_alias_map` and if so recursively update it so that the decorator can be used on sub-classes.

Reply:
> if so recursively update it

What you really want here is a `collections.ChainMap`, but it's not there in python 2.7.

Reply:
Or maybe
```
@staticmethod
def _get_alias_map(self):
    d = {}
    for cls in self.mro():
        d.update(cls.__dict__.get('_alias_map',{})
```

Reply:
I don't think ChainMap helps here, ex

```python

@_define_aliases({'linewidth': ['lw']})
class A:
   ...

@_define_aliases({'linewidth': ['linewidths', 'lws']})
class B(A):
   ...
```

You would want `B._alias_map['linewidth'] == ['lw', 'linewidths', 'lws']`

Reply:
Good point. Updated my other comment to reflect that `update` doesn't work either